check_limits.py

```diff
@@ -1,18 +1,40 @@
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+class Reporter:
+    def report(self, parameter, breach):
+        print(f"{parameter} is {breach}!")
 
-  return True
+class BatteryChecker:
+    def __init__(self, reporter=None):
+        self.reporter = reporter or Reporter()
```

Comment:
Good Use of Dependency Injection , but Reporter coupled with BatteryChecker, where Reporter is specific to terminal .
Refactor code for abstraction , make use of interface